src/provider/mod.rs

```diff
@@ -59,13 +324,193 @@ where
+    fn get_raw_block_transactions(
+        &self,
+        block_number: u64,
+    ) -> ProviderCall<T, (u64,), Vec<Transaction>> {
+        self.client()
+            .request("zks_getRawBlockTransactions", (block_number,))
+            .into()
+    }
+
+    /// Retrieves details for a given L1 batch.
+    fn get_l1_batch_details(
+        &self,
+        l1_batch_number: u64,
+    ) -> ProviderCall<T, (u64,), Option<L1BatchDetails>> {
+        self.client()
+            .request("zks_getL1BatchDetails", (l1_batch_number,))
+            .into()
+    }
+
+    /// Retrieves the bytecode of a transaction by its hash.
+    fn get_bytecode_by_hash(&self, tx_hash: B256) -> ProviderCall<T, (B256,), Option<Bytes>> {
+        self.client()
+            .request("zks_getBytecodeByHash", (tx_hash,))
+            .into()
+    }
+
+    /// Returns the range of blocks contained within a batch given by the batch number.
+    fn get_l1_batch_block_range(
+        &self,
+        l1_batch_number: u64,
+    ) -> ProviderCall<T, (u64,), Option<(U64, U64)>> {
+        self.client()
+            .request("zks_getL1BatchBlockRange", (l1_batch_number,))
+            .into()
+    }
+
+    /// Retrieves the current L1 gas price.
+    fn get_l1_gas_price(&self) -> ProviderCall<T, NoParams, U256> {
+        self.client().request_noparams("zks_getL1GasPrice").into()
+    }
+
+    /// Retrieves the current fee parameters.
+    fn get_fee_params(&self) -> ProviderCall<T, NoParams, FeeParams> {
+        self.client().request_noparams("zks_getFeeParams").into()
+    }
+
+    /// Retrieves the current fee parameters.
+    fn get_protocol_version(
+        &self,
+        version_id: Option<u16>,
+    ) -> ProviderCall<T, (Option<u16>,), Option<ProtocolVersion>> {
+        self.client()
+            .request("zks_getProtocolVersion", (version_id,))
+            .into()
+    }
+
+    /// Generates Merkle proofs for one or more storage values associated with a specific account,
+    /// accompanied by a proof of their authenticity. It verifies that these values remain unaltered.
+    fn get_proof(
+        &self,
+        address: Address,
+        keys: Vec<B256>,
+        l1_batch_number: u64,
+    ) -> ProviderCall<T, (Address, Vec<B256>, u64), Option<Proof>> {
+        self.client()
+            .request("zks_getProof", (address, keys, l1_batch_number))
+            .into()
+    }
+
+    /// Executes a transaction and returns its hash, storage logs, and events that
+    /// would have been generated if the transaction had already been included in the block.
+    /// The API has a similar behaviour to eth_sendRawTransaction
+    /// but with some extra data returned from it.
+    fn send_raw_transaction_with_detailed_output(
+        &self,
+        tx_bytes: Bytes,
+    ) -> ProviderCall<T, (Bytes,), TransactionDetailedResult> {
+        self.client()
+            .request("zks_sendRawTransactionWithDetailedOutput", (tx_bytes,))
+            .into()
+    }
```

Comment:
I can anticipate that this method will be deprecated in the future, so let's not add it for now as well.

Reply:
Got it, removed.